### PR TITLE
logpolicy: do not upload logs in tests

### PR DESCRIPTION
**Pull request by maisem**
Fixes tailscale/corp#10030

### CODE REVIEW

**Review by mihaip — APPROVED** [review]

**Review by dsnet — APPROVED** [review]
I wish GitHub UI would tell me that the entire body of `DialContext` was just moved from elsewhere.

File: log/sockstatlog/logger_test.go [hunk 1]
```diff
@@ -8,8 +8,29 @@ import (
...
 	"github.com/google/go-cmp/cmp"
 	"tailscale.com/net/sockstats"
+	"tailscale.com/tstest"
+	"tailscale.com/types/logger"
+	"tailscale.com/types/logid"
 )
 
+func TestResourceCleanup(t *testing.T) {
+	if !sockstats.IsAvailable {
+		t.Skip("sockstats not available")
```
[the remark this answers is not in the document]
Comment (maisem) [reply log/sockstatlog/logger_test.go:18]: Using `./tool/go` injects `tailscale_go` from what I see here
https://github.com/tailscale/tailscale/blob/33b359642e79fb04c90e6c92f516ec7b5a777a61/tool/gocross/gocross.go#L77-L81

https://github.com/tailscale/tailscale/blob/33b359642e79fb04c90e6c92f516ec7b5a777a61/tool/gocross/autoflags.go#L40

[the remark this answers is not in the document]
Comment (maisem) [reply log/sockstatlog/logger_test.go:18]: Do we not use `./tool/go` in CI?

[the remark this answers is not in the document]
Comment (maisem) [reply log/sockstatlog/logger_test.go:18]: Looks like it is
https://github.com/tailscale/tailscale/blob/9e81db50f6018f033776624514a50eb13d5a032e/.github/workflows/test.yml#L77-L82

**Pull request merged**